src/prefect/utilities/importtools.py

```diff
@@ -93,8 +93,7 @@ def load_script_as_module(path: str) -> ModuleType:
     parent_path = str(Path(path).resolve().parent)
     working_directory = os.getcwd()
 
-    # Generate unique module name for thread safety
-    module_name = f"__prefect_loader_{id(path)}__"
+    module_name = os.path.splitext(Path(path).name)[0]
```

Comment:
i'm realizing it may be worthwhile to check if this `module_name` already exists in `sys.modules` and if so, fall back to the loader name to avoid any breaking changes for people with poor file naming conventions